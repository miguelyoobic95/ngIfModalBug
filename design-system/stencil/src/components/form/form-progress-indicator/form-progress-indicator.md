---
name: Progress Indicator
category: Form Components
---

A simple progress indicator to show the user it's current progress in completing a form. It has a maximum of 7 visible steps. Additional steps are collapsed into a context menu on the penultimate element.

## Attributes

|Name|Type|Default|Description|
|---|---|---|---|
|`steps`|string[]|   |the steps to display in the indicator|
|`currentStep`|number| |index-based number to indicate the current step|
|`isCompleted`|boolean|`false`|True to display a completed progress indicator|

```yoo-form-progress-indicator.html
<div style="padding:10px;">
    <yoo-form-progress-indicator id="progress1" current-step=1></yoo-form-progress-indicator>
    <br/>
    <br/>
    <yoo-form-progress-indicator id="progress2" current-step=3></yoo-form-progress-indicator>
    <br/>
    <br/>
    <yoo-form-progress-indicator id="progress3" current-step=5></yoo-form-progress-indicator>
    <br>
    <br>
    <yoo-form-progress-indicator id="progress4" current-step=8></yoo-form-progress-indicator>
    <br/>
    <br/>
    <yoo-form-progress-indicator id="progress5" is-completed="true"></yoo-form-progress-indicator>
</div>
```

```yoo-form-progress-indicator.js
    var comp1 = document.querySelector('#progress1');
    comp1.steps = ["Step 1", "Step 2"];
    var comp2 = document.querySelector('#progress2');
    comp2.steps = ["Step 1", "Step 2", "Step 3", "Step 4", "Step 5", "Step 6", "Step 7"];
    var comp3 = document.querySelector('#progress3');
    comp3.steps = ["Step 1", "Step 2", "Step 3", "Step 4", "Step 5","Step 6", "Step 7", "Step 8", "Step 9"];
    var comp4 = document.querySelector('#progress4');
    comp4.steps = ["Step 1", "Step 2", "Step 3", "Step 4", "Step 5","Step 6", "Step 7", "Step 8", "Step 9"];
    var comp5 = document.querySelector('#progress5');
    comp5.steps = ["Step 1", "Step 2", "Step 3", "Step 4"];
```

## CSS

CSS classes can be used to change the color of the current step (default `accent`). The `class='vertical'` can be used to orient the element vertically.

|Type|Name|Description|
|---|---|---|
|Success|`success`|yoo-form-progress-indicator success|
|Danger|`danger`|yoo-form-progress-indicator danger|
|Info|`info`|yoo-form-progress-indicator info|
|Warning|`warning`|yoo-form-progress-indicator warning|
|Gradient Accent|`gradient-accent`|yoo-form-progress-indicator gradient-accent|
|Gradient Dark|`gradient-dark`|yoo-form-progress-indicator gradient-dark|
|Gradient Success|`gradient-success`|yoo-form-progress-indicator gradient-success|
|Gradient Danger|`gradient-danger`|yoo-form-progress-indicator gradient-danger|
|Gradient Info|`gradient-info`|yoo-form-progress-indicator gradient-info|
|Gradient Warning|`gradient-warning`|yoo-form-progress-indicator gradient-warning|
|Vertical|`vertical`|yoo-form-progress-indicator vertical|

```yoo-form-progress-indicator-classes.html
<div style="padding:10px;">
    <yoo-form-progress-indicator class="vertical" current-step=3></yoo-form-progress-indicator>
    <br/>
    <br/>
    <yoo-form-progress-indicator class="success" current-step=0></yoo-form-progress-indicator>
    <br/>
    <br/>
    <yoo-form-progress-indicator class="danger" current-step=1></yoo-form-progress-indicator>
    <br/>
    <br/>
    <yoo-form-progress-indicator class="info" current-step=2></yoo-form-progress-indicator>
    <br/>
    <br/>
    <yoo-form-progress-indicator class="warning" current-step=3></yoo-form-progress-indicator>
    <br/>
    <br/>
    <yoo-form-progress-indicator class="gradient-accent" current-step=4></yoo-form-progress-indicator>
    <br/>
    <br/>
    <yoo-form-progress-indicator class="gradient-dark" current-step=5></yoo-form-progress-indicator>
    <br/>
    <br/>
    <yoo-form-progress-indicator class="gradient-success" current-step=6></yoo-form-progress-indicator>
    <br/>
    <br/>
    <yoo-form-progress-indicator class="gradient-danger" current-step=7></yoo-form-progress-indicator>
    <br/>
    <br/>
    <yoo-form-progress-indicator class="gradient-info" current-step=4></yoo-form-progress-indicator>
    <br/>
    <br/>
    <yoo-form-progress-indicator class="gradient-warning" current-step=3></yoo-form-progress-indicator>
</div>
```

```yoo-form-progress-indicator-classes.js
    var comps = document.querySelectorAll('yoo-form-progress-indicator');
    comps.forEach(comp => comp.steps = ["Step 1", "Step 2", "Step 3", "Step 4", "Step 5", "Step 6", "Step 7", "Step 8", "Step 9"]);
```

## Events

|Attr|Description|
|---|---|---|
|`stepSelected`|triggered when clicking on a step, it returns that step|