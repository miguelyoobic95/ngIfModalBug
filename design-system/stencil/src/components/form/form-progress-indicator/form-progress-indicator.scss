

@mixin current-color($color) {
    .progress-indicator-container {
        .step-container {
            .step-circle {
                &.current {
                    background-color: cssvar($color);
                }
            }
        }
    }
}

@mixin current-gradient($gradient) {
    .progress-indicator-container {
        .step-container {
            .step-circle {
                &.current {
                    background: cssvar($gradient);
                }
            }
        }
    }
}

:host {
    yoo-context-menu {
        /deep/.dropdown-content {
            width: 11.25rem;
        }
    }
}

:host {
    .progress-indicator-container {
        position: relative;
        flex-direction: row;
        .step-container {
            cursor: pointer;
            padding-right: 4.6875rem;
            .step-title {
                //color: cssvar(dark-20);
                @include text-truncate();
                min-width: 2rem;
                width: 3.4375rem;
                font-size: 1rem;
                padding-bottom: 1em;
                text-align: left;
            }
            .step-circle {
                @include border-radius(50%);
                background-color: cssvar(dark-10);
                color: cssvar(light);
                font-size: 0.875rem;
                padding: 4.5px 11.5px;
                height: 30px;
                width: 30px;
                &.more {
                    padding: 5.5px 8.5px;
                    cursor: context-menu;
                }
                &.completed {
                    background-color: cssvar(dark-20);
                    padding: 4.5px 0.5rem;
                    color: cssvar(accent);
                }
                &.current {
                    background-color: cssvar(accent);
                }
            }
            .context-container {
                
                span {
                    padding: 0.5rem 1rem;
                    line-height: 1.5;
                    width: 100%;
                    position: relative;
                    border-bottom: 1px solid cssvar(dark-20);
                    &:last-child {
                        border-bottom: none;
                    }
                    &:hover {
                        background: cssvar(accent-05);
                        color: cssvar(dark);
                    }
                    &.context-current {
                        font-weight: bold;
                    }
                    i {
                        position: absolute;
                        top: 12px;
                        right: 10px;
                    }
                }
            }
        }
        .progress-line {
            position: absolute;
            top: 3.3125rem;
            margin-left: 1px;
            z-index: -1;
            height: 5px;
            background-color: cssvar(dark-20);
            width: 8.45rem;
            &.faded {
                background-color: cssvar(dark-10);
                z-index: -2;
            }
        }
    }
}

:host(.vertical) {
    .progress-indicator-container {
        flex-direction: column;
        max-width: 2rem;
        .step-container {
            position: relative;
            padding-bottom: 2.8125rem;
            padding-right: 0rem;
            .step-title {
                position: relative;
                top: 2.3125rem;
                bottom: 0px;
                left: 2.8125rem;
                right: 0px;
            }
            .step-circle {
                text-align: center;
                padding: 4.5px 0.5px;
                &.completed {
                    .completed-icon {
                        text-align: center;
                    }
                }
            }
        }
        .progress-line {
            position: absolute;
            top: 2.6875rem;
            right: 14px;
            z-index: -1;
            width: 5px;
            background-color: cssvar(dark-20);
            height: 8.45rem;
            &.faded {
                background-color: cssvar(dark-10);
                z-index: -2;
            }
        }
    }
}

// Colors
:host(.success) {
    .progress-indicator-container .step-container .step-circle.completed {
        color: cssvar(success);
    }
    @include current-color(success);
}

:host(.danger) {
    .progress-indicator-container .step-container .step-circle.completed {
        color: cssvar(danger);
    }
    @include current-color(danger);
}

:host(.warning) {
    .progress-indicator-container .step-container .step-circle.completed {
        color: cssvar(warning);
    }
    @include current-color(warning);
}

:host(.info) {
    .progress-indicator-container .step-container .step-circle.completed {
        color: cssvar(info);
    }
    @include current-color(info);
}

:host(.gradient-accent) {
    .progress-indicator-container .step-container .step-circle.completed {
        color: cssvar(gradient-accent);
    }
    @include current-gradient(gradient-accent);
}

:host(.gradient-dark) {
    .progress-indicator-container .step-container .step-circle.completed {
        color: cssvar(gradient-dark);
    }
    @include current-gradient(gradient-dark);
}

:host(.gradient-success) {
    .progress-indicator-container .step-container .step-circle.completed {
        color: cssvar(gradient-success);
    }
    @include current-gradient(gradient-success);
}

:host(.gradient-danger) {
    .progress-indicator-container .step-container .step-circle.completed {
        color: cssvar(gradient-danger);
    }
    @include current-gradient(gradient-danger);
}

:host(.gradient-info) {
    .progress-indicator-container .step-container .step-circle.completed {
        color: cssvar(gradient-info);
    }
    @include current-gradient(gradient-info);
}

:host(.gradient-warning) {
    .progress-indicator-container .step-container .step-circle.completed {
        color: cssvar(gradient-warning);
    }
    @include current-gradient(gradient-warning);
}