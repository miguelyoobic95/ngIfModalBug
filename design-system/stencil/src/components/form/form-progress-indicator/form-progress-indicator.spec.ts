import { TestWindow } from '@stencil/core/testing';

let mockDimensions = jest.fn();
jest.mock('../../../utils/helpers', () => {
    return {
        getElementDimensions: mockDimensions
    };
});
import { YooFormProgressIndicatorComponent } from './form-progress-indicator';

describe('YooFormProgressIndicatorComponent', () => {
    let window;
    beforeEach(() => {
      window = new TestWindow();
    });
    it('should build', () => {
        expect(new YooFormProgressIndicatorComponent()).toBeTruthy();
    });

    describe('Status Rendering', () => {
        beforeEach( async () => {
            mockDimensions.mockImplementation(() => {
                return {width: 1000, height: 100};
            });
        });
        it('Should have progress steps', async () => {
            let element = await window.load({
                components: [YooFormProgressIndicatorComponent],
                html: '<yoo-form-progress-indicator></yoo-form-progress-indicator>'
            });
            element.steps = ['Step 1', 'Step 2'];
            await window.flush();
            expect(element).toMatchSnapshot();
        });
        it('Should have a current step', async () => {
            let element = await window.load({
                components: [YooFormProgressIndicatorComponent],
                html: '<yoo-form-progress-indicator current-step=1></yoo-form-progress-indicator>'
            });
            element.steps = ['Step 1', 'Step 2'];
            await window.flush();
            expect(element).toMatchSnapshot();
        });
        it('Should be fully completed', async () => {
            let element = await window.load({
                components: [YooFormProgressIndicatorComponent],
                html: '<yoo-form-progress-indicator is-completed="true"></yoo-form-progress-indicator>'
            });
            expect(element).toMatchSnapshot();
        });
    });

    describe('Step rendering', async () => {
        beforeEach( async () => {
            mockDimensions.mockImplementation(() => {
                return {width: 1000, height: 100};
            });
        });
        it('Should render the correct number of steps', async () => {
            let element = await window.load({
                components: [YooFormProgressIndicatorComponent],
                html: '<yoo-form-progress-indicator current-step=1></yoo-form-progress-indicator>'
            });
            element.steps = ['Step 1', 'Step 2'];
            await window.flush();
            expect(element.getElementsByClassName('step-container')).toHaveLength(2);
            expect(element).toMatchSnapshot();
        });
        it('Should only render a maximum of 7 step-containers', async () => {
            let element = await window.load({
                components: [YooFormProgressIndicatorComponent],
                html: '<yoo-form-progress-indicator current-step=1></yoo-form-progress-indicator>'
            });
            element.steps = ['Step 1', 'Step 2', 'Step 3', 'Step 4', 'Step 5', 'Step 6', 'Step7', 'Step 8'];
            await window.flush();
            expect(element.getElementsByClassName('step-container')).toHaveLength(7);
            expect(element).toMatchSnapshot();
        });
        it('Should be completed after toggling the completed flag', async () => {
            let element = await window.load({
                components: [YooFormProgressIndicatorComponent],
                html: '<yoo-form-progress-indicator></yoo-form-progress-indicator>'
            });
            expect(element.isCompleted).toBeFalsy();
            element.steps = ['Step 1', 'Step 2'];
            element.isCompleted = true;
            await window.flush();
            expect(element.getElementsByClassName('completed')).toHaveLength(2);
            expect(element).toMatchSnapshot();
        });
        it('Should be completed after toggling the completed flag and while having more than 7 items', async () => {
            let element = await window.load({
                components: [YooFormProgressIndicatorComponent],
                html: '<yoo-form-progress-indicator></yoo-form-progress-indicator>'
            });
            expect(element.isCompleted).toBeFalsy();
            element.steps = ['Step 1', 'Step 2', 'Step 3', 'Step 4', 'Step 5', 'Step 6', 'Step7', 'Step 8'];
            element.isCompleted = true;
            await window.flush();
            expect(element.getElementsByClassName('completed')).toHaveLength(7);
            expect(element).toMatchSnapshot();
        });
    });

    describe('Step rendering according to parent', async () => {
        it('Should render less steps', async () => {
            mockDimensions.mockImplementation(() => {
                return {width: 200, height: 100};
            });

            let element = await window.load({
                components: [YooFormProgressIndicatorComponent],
                html: '<yoo-form-progress-indicator current-step=1></yoo-form-progress-indicator>'
            });
            element.steps = ['Step 1', 'Step 2', 'Step 3', 'Step 4', 'Step 5', 'Step 6', 'Step7', 'Step 8'];
            await window.flush();
            expect(element.getElementsByClassName('step-container')).toHaveLength(2);

            expect(element).toMatchSnapshot();
        });
    });

  describe('events', () => {
    let element, res, emitEvent;
    beforeEach( async () => {
      mockDimensions.mockImplementation(() => {
        return {width: 200, height: 100};
      });

      element = await window.load({
        components: [YooFormProgressIndicatorComponent],
        html: '<yoo-form-progress-indicator current-step=1></yoo-form-progress-indicator>'
      });
      element.steps = ['Step 1', 'Step 2', 'Step 3', 'Step 4', 'Step 5', 'Step 6', 'Step7', 'Step 8'];
      await window.flush();
      emitEvent = jest.fn(ev => { res = ev.detail; });
    });
    it ('should emit stepSelected on click', async () => {
      window.document.addEventListener('stepSelected', emitEvent);
      let span = element.querySelector('.context-container > span');
      await span.click();
      expect(emitEvent).toHaveBeenCalled();
      expect(res).toEqual(element.steps[0]);
      // expect(emitEvent).toHaveEventData([element.steps[0]]);
    });
  });
});
