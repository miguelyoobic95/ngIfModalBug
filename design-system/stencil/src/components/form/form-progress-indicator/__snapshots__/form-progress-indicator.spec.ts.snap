// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooFormProgressIndicatorComponent Status Rendering Should be fully completed 1`] = `
<yoo-form-progress-indicator
  class="hydrated"
  data-yoo-form-progress-indicator-host=""
  is-completed="true"
>
  
  <div
    attr-layout="row"
    class="progress-indicator-container"
    data-yoo-form-progress-indicator=""
  />
</yoo-form-progress-indicator>
`;

exports[`YooFormProgressIndicatorComponent Status Rendering Should have a current step 1`] = `
<yoo-form-progress-indicator
  class="hydrated"
  current-step="1"
  data-yoo-form-progress-indicator-host=""
>
  
  <div
    attr-layout="row"
    class="progress-indicator-container"
    data-yoo-form-progress-indicator=""
  >
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 1
        </span>
      </div>
      <div
        class="step-circle completed"
        data-yoo-form-progress-indicator=""
      >
        <span
          class="completed-icon"
          data-yoo-form-progress-indicator=""
        >
          <i
            class="yo-check"
            data-yoo-form-progress-indicator=""
          />
        </span>
      </div>
      <div
        class="progress-line"
        data-yoo-form-progress-indicator=""
      />
    </div>
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 2
        </span>
      </div>
      <div
        class="step-circle current"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          2
        </span>
      </div>
      
    </div>
  </div>
</yoo-form-progress-indicator>
`;

exports[`YooFormProgressIndicatorComponent Status Rendering Should have progress steps 1`] = `
<yoo-form-progress-indicator
  class="hydrated"
  data-yoo-form-progress-indicator-host=""
>
  
  <div
    attr-layout="row"
    class="progress-indicator-container"
    data-yoo-form-progress-indicator=""
  >
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 1
        </span>
      </div>
      <div
        class="step-circle"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          1
        </span>
      </div>
      <div
        class="progress-line faded"
        data-yoo-form-progress-indicator=""
      />
    </div>
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 2
        </span>
      </div>
      <div
        class="step-circle"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          2
        </span>
      </div>
      
    </div>
  </div>
</yoo-form-progress-indicator>
`;

exports[`YooFormProgressIndicatorComponent Step rendering Should be completed after toggling the completed flag 1`] = `
<yoo-form-progress-indicator
  class="hydrated"
  data-yoo-form-progress-indicator-host=""
>
  
  <div
    attr-layout="row"
    class="progress-indicator-container"
    data-yoo-form-progress-indicator=""
  >
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 1
        </span>
      </div>
      <div
        class="step-circle completed"
        data-yoo-form-progress-indicator=""
      >
        <span
          class="completed-icon"
          data-yoo-form-progress-indicator=""
        >
          <i
            class="yo-check"
            data-yoo-form-progress-indicator=""
          />
        </span>
      </div>
      <div
        class="progress-line"
        data-yoo-form-progress-indicator=""
      />
    </div>
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 2
        </span>
      </div>
      <div
        class="step-circle completed"
        data-yoo-form-progress-indicator=""
      >
        <span
          class="completed-icon"
          data-yoo-form-progress-indicator=""
        >
          <i
            class="yo-check"
            data-yoo-form-progress-indicator=""
          />
        </span>
      </div>
      
    </div>
  </div>
</yoo-form-progress-indicator>
`;

exports[`YooFormProgressIndicatorComponent Step rendering Should be completed after toggling the completed flag and while having more than 7 items 1`] = `
<yoo-form-progress-indicator
  class="hydrated"
  data-yoo-form-progress-indicator-host=""
>
  
  <div
    attr-layout="row"
    class="progress-indicator-container"
    data-yoo-form-progress-indicator=""
  >
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 1
        </span>
      </div>
      <div
        class="step-circle completed"
        data-yoo-form-progress-indicator=""
      >
        <span
          class="completed-icon"
          data-yoo-form-progress-indicator=""
        >
          <i
            class="yo-check"
            data-yoo-form-progress-indicator=""
          />
        </span>
      </div>
      <div
        class="progress-line"
        data-yoo-form-progress-indicator=""
      />
    </div>
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 2
        </span>
      </div>
      <div
        class="step-circle completed"
        data-yoo-form-progress-indicator=""
      >
        <span
          class="completed-icon"
          data-yoo-form-progress-indicator=""
        >
          <i
            class="yo-check"
            data-yoo-form-progress-indicator=""
          />
        </span>
      </div>
      <div
        class="progress-line"
        data-yoo-form-progress-indicator=""
      />
    </div>
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 3
        </span>
      </div>
      <div
        class="step-circle completed"
        data-yoo-form-progress-indicator=""
      >
        <span
          class="completed-icon"
          data-yoo-form-progress-indicator=""
        >
          <i
            class="yo-check"
            data-yoo-form-progress-indicator=""
          />
        </span>
      </div>
      <div
        class="progress-line"
        data-yoo-form-progress-indicator=""
      />
    </div>
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 4
        </span>
      </div>
      <div
        class="step-circle completed"
        data-yoo-form-progress-indicator=""
      >
        <span
          class="completed-icon"
          data-yoo-form-progress-indicator=""
        >
          <i
            class="yo-check"
            data-yoo-form-progress-indicator=""
          />
        </span>
      </div>
      <div
        class="progress-line"
        data-yoo-form-progress-indicator=""
      />
    </div>
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 5
        </span>
      </div>
      <div
        class="step-circle completed"
        data-yoo-form-progress-indicator=""
      >
        <span
          class="completed-icon"
          data-yoo-form-progress-indicator=""
        >
          <i
            class="yo-check"
            data-yoo-form-progress-indicator=""
          />
        </span>
      </div>
      <div
        class="progress-line"
        data-yoo-form-progress-indicator=""
      />
    </div>
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 6
        </span>
      </div>
      <yoo-context-menu
        data-yoo-form-progress-indicator=""
      >
        <div
          class="step-circle completed"
          data-yoo-form-progress-indicator=""
          slot="trigger"
        >
          <span
            class="completed-icon"
            data-yoo-form-progress-indicator=""
          >
            <i
              class="yo-more"
              data-yoo-form-progress-indicator=""
            />
          </span>
        </div>
        <div
          attr-layout="column"
          class="context-container"
          data-yoo-form-progress-indicator=""
        >
          <span
            data-yoo-form-progress-indicator=""
          >
            Step 6
            <i
              class="yo-check"
              data-yoo-form-progress-indicator=""
            />
          </span>
          <span
            data-yoo-form-progress-indicator=""
          >
            Step7
            <i
              class="yo-check"
              data-yoo-form-progress-indicator=""
            />
          </span>
        </div>
      </yoo-context-menu>
      <div
        class="progress-line"
        data-yoo-form-progress-indicator=""
      />
    </div>
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 8
        </span>
      </div>
      <div
        class="step-circle completed"
        data-yoo-form-progress-indicator=""
      >
        <span
          class="completed-icon"
          data-yoo-form-progress-indicator=""
        >
          <i
            class="yo-check"
            data-yoo-form-progress-indicator=""
          />
        </span>
      </div>
      
    </div>
  </div>
</yoo-form-progress-indicator>
`;

exports[`YooFormProgressIndicatorComponent Step rendering Should only render a maximum of 7 step-containers 1`] = `
<yoo-form-progress-indicator
  class="hydrated"
  current-step="1"
  data-yoo-form-progress-indicator-host=""
>
  
  <div
    attr-layout="row"
    class="progress-indicator-container"
    data-yoo-form-progress-indicator=""
  >
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 1
        </span>
      </div>
      <div
        class="step-circle completed"
        data-yoo-form-progress-indicator=""
      >
        <span
          class="completed-icon"
          data-yoo-form-progress-indicator=""
        >
          <i
            class="yo-check"
            data-yoo-form-progress-indicator=""
          />
        </span>
      </div>
      <div
        class="progress-line"
        data-yoo-form-progress-indicator=""
      />
    </div>
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 2
        </span>
      </div>
      <div
        class="step-circle current"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          2
        </span>
      </div>
      <div
        class="progress-line faded"
        data-yoo-form-progress-indicator=""
      />
    </div>
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 3
        </span>
      </div>
      <div
        class="step-circle"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          3
        </span>
      </div>
      <div
        class="progress-line faded"
        data-yoo-form-progress-indicator=""
      />
    </div>
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 4
        </span>
      </div>
      <div
        class="step-circle"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          4
        </span>
      </div>
      <div
        class="progress-line faded"
        data-yoo-form-progress-indicator=""
      />
    </div>
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 5
        </span>
      </div>
      <div
        class="step-circle"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          5
        </span>
      </div>
      <div
        class="progress-line faded"
        data-yoo-form-progress-indicator=""
      />
    </div>
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 6
        </span>
      </div>
      <yoo-context-menu
        data-yoo-form-progress-indicator=""
      >
        <div
          class="step-circle more"
          data-yoo-form-progress-indicator=""
          slot="trigger"
        >
          <span
            data-yoo-form-progress-indicator=""
          >
            <i
              class="yo-more"
              data-yoo-form-progress-indicator=""
            />
          </span>
        </div>
        <div
          attr-layout="column"
          class="context-container"
          data-yoo-form-progress-indicator=""
        >
          <span
            class=""
            data-yoo-form-progress-indicator=""
          >
            Step 6
          </span>
          <span
            class=""
            data-yoo-form-progress-indicator=""
          >
            Step7
          </span>
        </div>
      </yoo-context-menu>
      <div
        class="progress-line faded"
        data-yoo-form-progress-indicator=""
      />
    </div>
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 8
        </span>
      </div>
      <div
        class="step-circle"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          8
        </span>
      </div>
      
    </div>
  </div>
</yoo-form-progress-indicator>
`;

exports[`YooFormProgressIndicatorComponent Step rendering Should render the correct number of steps 1`] = `
<yoo-form-progress-indicator
  class="hydrated"
  current-step="1"
  data-yoo-form-progress-indicator-host=""
>
  
  <div
    attr-layout="row"
    class="progress-indicator-container"
    data-yoo-form-progress-indicator=""
  >
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 1
        </span>
      </div>
      <div
        class="step-circle completed"
        data-yoo-form-progress-indicator=""
      >
        <span
          class="completed-icon"
          data-yoo-form-progress-indicator=""
        >
          <i
            class="yo-check"
            data-yoo-form-progress-indicator=""
          />
        </span>
      </div>
      <div
        class="progress-line"
        data-yoo-form-progress-indicator=""
      />
    </div>
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 2
        </span>
      </div>
      <div
        class="step-circle current"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          2
        </span>
      </div>
      
    </div>
  </div>
</yoo-form-progress-indicator>
`;

exports[`YooFormProgressIndicatorComponent Step rendering according to parent Should render less steps 1`] = `
<yoo-form-progress-indicator
  class="hydrated"
  current-step="1"
  data-yoo-form-progress-indicator-host=""
>
  
  <div
    attr-layout="row"
    class="progress-indicator-container"
    data-yoo-form-progress-indicator=""
  >
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 1
        </span>
      </div>
      <yoo-context-menu
        data-yoo-form-progress-indicator=""
      >
        <div
          class="step-circle current more"
          data-yoo-form-progress-indicator=""
          slot="trigger"
        >
          <span
            data-yoo-form-progress-indicator=""
          >
            <i
              class="yo-more"
              data-yoo-form-progress-indicator=""
            />
          </span>
        </div>
        <div
          attr-layout="column"
          class="context-container"
          data-yoo-form-progress-indicator=""
        >
          <span
            class=""
            data-yoo-form-progress-indicator=""
          >
            Step 1
            <i
              class="yo-check"
              data-yoo-form-progress-indicator=""
            />
          </span>
          <span
            class="context-current"
            data-yoo-form-progress-indicator=""
          >
            Step 2
          </span>
          <span
            class=""
            data-yoo-form-progress-indicator=""
          >
            Step 3
          </span>
          <span
            class=""
            data-yoo-form-progress-indicator=""
          >
            Step 4
          </span>
          <span
            class=""
            data-yoo-form-progress-indicator=""
          >
            Step 5
          </span>
          <span
            class=""
            data-yoo-form-progress-indicator=""
          >
            Step 6
          </span>
          <span
            class=""
            data-yoo-form-progress-indicator=""
          >
            Step7
          </span>
        </div>
      </yoo-context-menu>
      <div
        class="progress-line faded"
        data-yoo-form-progress-indicator=""
      />
    </div>
    <div
      class="step-container"
      data-yoo-form-progress-indicator=""
    >
      <div
        class="step-title"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          Step 8
        </span>
      </div>
      <div
        class="step-circle"
        data-yoo-form-progress-indicator=""
      >
        <span
          data-yoo-form-progress-indicator=""
        >
          8
        </span>
      </div>
      
    </div>
  </div>
</yoo-form-progress-indicator>
`;
